src/Microsoft.Azure.WebJobs.Host/Blobs/Listeners/ScanBlobScanLogHybridPollingStrategy.cs

```diff
@@ -223,10 +228,13 @@ private void ThrowIfDisposed()
             // record continuation token for next chunk retrieval
             containerScanInfo.ContinuationToken = blobSegment.ContinuationToken;
 
-            // if ending a cycle then copy currentSweepCycleStartTime to lastSweepCycleStartTime
-            if (blobSegment.ContinuationToken == null)
+            // if ending a cycle then copy currentSweepCycleStartTime to lastSweepCycleStartTime, if changed
+            if (blobSegment.ContinuationToken == null &&
+                containerScanInfo.CurrentSweepCycleLatestModified > containerScanInfo.LastSweepCycleLatestModified)
             {
                 containerScanInfo.LastSweepCycleLatestModified = containerScanInfo.CurrentSweepCycleLatestModified;
+                await _blobScanInfoManager.UpdateLatestScanAsync(container.ServiceClient.Credentials.AccountName,
+                    container.Name, containerScanInfo.LastSweepCycleLatestModified);
```

Comment:
@mathewc  -- is the full scan at startup considered our failsafe? This persistence may not be a good idea, then, if it's removing that. I had assumed that the log scanning was fullproof (within 2 hours) and thus was the failsafe here. But it sounds like that may not be the case if logs aren't guaranteed.

I was even thinking that, on startup, we could check the last updated time of the scanInfo blob and delete it if it were older than 2 hours so we'd only have to do a full scan if we think we may have blobs added outside of that 2 hour window.

If we do, indeed, need that full scan at startup, maybe I can see if there's a way to trim down that 140ms per-blob hit we take while we scan. Parallelize it, try to reduce network calls, etc.